packages/app-utils/src/components/AuthBootstrap.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { AuthPluginComponent } from '@deephaven/auth-plugin';
+import {
+  AuthPluginAnonymous,
+  AuthPluginParent,
+  AuthPluginPsk,
+} from '@deephaven/auth-core-plugins';
+import { LoadingOverlay } from '@deephaven/components';
+import { PluginsContext } from './PluginsBootstrap';
+import { getAuthPluginComponent } from '../plugins';
+import useClient from './useClient';
+
+export type AuthConfigMap = Map<string, string>;
+
+export type AuthBootstrapProps = {
+  /**
+   * The children to render after authentication is completed.
+   */
+  children: React.ReactNode;
+};
+
+/** Core auth plugins that are always loaded */
+const CORE_AUTH_PLUGINS = new Map([
+  ['@deephaven/auth-core-plugins.AuthPluginPsk', AuthPluginPsk],
+  ['@deephaven/auth-core-plugins.AuthPluginParent', AuthPluginParent],
+  ['@deephaven/auth-core-plugins.AuthPluginAnonymous', AuthPluginAnonymous],
+]);
+
+/**
+ * AuthBootstrap component. Handles displaying the auth plugin and authenticating.
+ */
+export function AuthBootstrap({ children }: AuthBootstrapProps) {
+  const client = useClient();
+  const plugins = useContext(PluginsContext);
+  const [authConfig, setAuthConfig] = useState<AuthConfigMap>();
+  const [AuthComponent, setAuthComponent] = useState<AuthPluginComponent>();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [error, setError] = useState<unknown>();
+
+  useEffect(
+    function initAuthConfigValues() {
+      let isCancelled = false;
```

Comment:
Hmmm yea in this case we need to map it to a map, so it'd be something like:
```
useAsyncAffect(async function loadAuthConfigValues() {
  const newAuthConfigValues = await client.getAuthConfigValues();
  return new Map(newAuthConfigValues);
}, setAuthConfig, setError, [client])
```
I like that though, makes this paradigm much easier.

Reply:
Did you want to add that hook in this change or in another since this change is already pretty big? I saw that `return () => isCancelled = true` several times in this PR and wasn't sure if it's used elsewhere in our code right now